File: Module 3/3A Javascript/JavaScript Coding Challenge/Task2.js
// JavaScript Coding Challenge

// Create a length converter function

function convertlength(value, fromUnit, toUnit) {
    let valueInMeter;

    //Convert from the original unit to meters
    if (fromUnit === "meters") {
        valueInMeter = value;
    } else if (fromUnit === "kilometers") {
        valueInMeter = value * 1000;
    } else if (fromUnit === "centimenters") {
        valueInMeter = value / 100;
    } else if (fromUnit === "millimeters") {
        valueInMeter = value / 1000;
    } else {
        return "invalid fromUnit";
    }

    // Convert from meters to the target unit
    if (toUnit === "meters") {
        return valueInMeter;
    } else if (toUnit === "kilometers") {
        return valueInMeter / 1000;
    } else if (toUnit === "centimeters") {
        return valueInMeter * 100;
    } else if (toUnit === "millimeters") {
        return valueInMeter * 1000;
    } else {
        return "invalid toUnit"; 
    }
}

// Sample usage:
let length = 15;
let fromUnit = "meters";
let toUnit = "kilometers";
console.log(length + " " + fromUnit + " is equal to " + convertlength(length, fromUnit, toUnit) + " " + toUnit);

fromUnit = "centimeters";
toUnit = "millimeters";
console.log(length + " " + fromUnit + " is equal to " + convertlength(length, fromUnit, toUnit) + " " + toUnit);

File: Module 4/4C_Everyday_Go_To_Python/Python_Coding_Challenge.py/Task2.py
"""
Create a length converter function
"""
def length_converter(value, from_unit, to_unit):
    # Conversion rates to meters
    conversion_to_meters = {
        "meters": 1,
        "kilometers": 1000,
        "miles": 1609.34,
        "feet": 0.3048,
        "inches": 0.0254,
    }
    
    # Ensure the units are valid
    if from_unit not in conversion_to_meters or to_unit not in conversion_to_meters:
        raise ValueError("Invalid units. Valid units are: meters, kilometers, miles, feet, inches.")

    # Convert the input value to meters
    value_in_meters = value * conversion_to_meters[from_unit]

    # Convert from meters to the desired unit
    converted_value = value_in_meters / conversion_to_meters[to_unit]

    return converted_value

# Example usage
value = 5  # Length to convert
from_unit = "meters"  # Unit to convert from
to_unit = "feet"  # Unit to convert to

result = length_converter(value, from_unit, to_unit)
print(f"{value} {from_unit} is equal to {result:.2f} {to_unit}")
